chalice/analyzer.py

```diff
@@ -126,7 +126,7 @@ def __init__(self, service_name):
 
     def __eq__(self, other):
         # type: (Any) -> bool
-        if not isinstance(other, self.__class__):
+        if not isinstance(other, Boto3ClientType):
```

Comment:
What was the reason for changing these from using `__class__`? It is unfortunate now that if we change the name of this class that we have to remember to change the `isinstance` check. I think this change happens a couple more times as well.

Reply:
Mypy bug: https://github.com/python/mypy/issues/3061

```
$ cat /tmp/a.py
from typing import Any


class A:
    def __init__(self, x):
        # type: (int) -> None
        self.x = x

    def __eq__(self, other):
        # type: (Any) -> bool
        if not isinstance(other, self.__class__):
            return False
        return self.x == other.x

$ mypy /tmp/a.py
/tmp/a.py:13: error: "object" has no attribute "x"
```

vs.

```
$ cat /tmp/a.py
from typing import Any


class A:
    def __init__(self, x):
        # type: (int) -> None
        self.x = x

    def __eq__(self, other):
        # type: (Any) -> bool
        if not isinstance(other, A):
            return False
        return self.x == other.x
$ mypy /tmp/a.py
$
```

Reply:
Sounds good, I'll add a comment about this before I merge.